modules/ximgproc/samples/disparity_filtering.cpp

```diff
@@ -4,162 +4,331 @@
+    "{dst_path       |None       | optional path to save the resulting filtered disparity map        }"
+    "{dst_raw_path   |None       | optional path to save raw disparity map before filtering          }"
+    "{algorithm      |bm         | stereo matching method (bm or sgbm)                               }"
+    "{filter         |wls_conf   | used post-filtering (wls_conf or wls_wo_conf)                     }"
+    "{no-display     |           | don't display results                                             }"
+    "{no-downscale   |           | prevent stereo matching on downscaled views                       }"
+    "{dst_conf_path  |None       | optional path to save the confidence map used in filtering        }"
+    "{vis_mult       |1.0        | coefficient used to scale disparity map visualizations            }"
+    "{max_disparity  |160        | parameter of stereo matching                                      }"
+    "{window_size    |19         | parameter of stereo matching                                      }"
+    "{wls_lambda     |4000.0     | parameter of post-filtering                                       }"
+    "{wls_sigma      |1.0        | parameter of post-filtering                                       }"
+    ;
+
+int main(int argc, char** argv)
 {
-    string name;
-    string dataset_folder;
-    string left_file,right_file,GT_file;
-    dataset_entry(string _dataset_folder): dataset_folder(_dataset_folder){}
-    void readEntry(Mat& dst_left,Mat& dst_right,Mat& dst_GT)
+    CommandLineParser parser(argc,argv,keys);
+    parser.about("Disparity Filtering Demo");
+    if (parser.has("help"))
     {
-        dst_left  = imread(dataset_folder+"/"+left_file,  IMREAD_COLOR);
-        dst_right = imread(dataset_folder+"/"+right_file, IMREAD_COLOR);
-        Mat raw_disp = imread(dataset_folder+"/"+GT_file, IMREAD_COLOR);
-        dst_GT = Mat(raw_disp.rows,raw_disp.cols,CV_16S);
-        for(int i=0;i<raw_disp.rows;i++)
-            for(int j=0;j<raw_disp.cols;j++)
-            {
-                Vec3b bgrPixel = raw_disp.at<Vec3b>(i, j);
-                dst_GT.at<short>(i,j) = 64*bgrPixel.val[2]+bgrPixel.val[1]/4; //16-multiplied disparity
-            }
+        parser.printMessage();
+        return 0;
     }
-};
 
-struct config
-{
-    Ptr<StereoMatcher> matcher_instance;
-    Ptr<DisparityFilter> filter_instance;
-    config(Ptr<StereoMatcher> _matcher_instance,Ptr<DisparityFilter> _filter_instance)
+    String left_im;
+    String right_im;
+    String GT_path;
+
+    if (argc<2)
     {
-        matcher_instance = _matcher_instance;
-        filter_instance = _filter_instance;
+        cout<<"Processing 'aloe' by default. Use -h for help"<<endl;
+        left_im = "../../../../opencv-master/samples/data/aloeL.jpg";
+        right_im = "../../../../opencv-master/samples/data/aloeR.jpg";
+        GT_path = "../../../../opencv-master/samples/data/aloeGT.png";
```

Comment:
It is better to set default paths in CommandLineParser keys initialization and use `../data/aloeL.jpg`